### ISSUE TITLE
Import layer to Spatialite/PostGis does not support binary data

### ISSUE THREAD

**Issue by audun** [qgis/QGIS#36705]
**Describe the bug**

The QGIS Spatialite provider does not support binary data fields.

**How to Reproduce**

1. Create a new scratch layer

![image](https://user-images.githubusercontent.com/454644/82819495-c1b45b80-9ea0-11ea-8e6f-79bc701a8469.png)

2. Add a binary field

![image](https://user-images.githubusercontent.com/454644/82819298-726e2b00-9ea0-11ea-9be1-678bb693eb61.png)

Result looks good so far:
![image](https://user-images.githubusercontent.com/454644/82819707-1d7ee480-9ea1-11ea-8578-3a21c9f06809.png)
The attached project file is saved at this point: [minimal_binary_field.zip](https://github.com/qgis/QGIS/files/4677779/minimal_binary_field.zip)

3. Create a new Spatialite db
4. Import the layer to the new DB

5. Observe, with the appropriate amount of sadness, the error message:

![image](https://user-images.githubusercontent.com/454644/82819965-99792c80-9ea1-11ea-9662-724c2bfcbc0f.png)

**QGIS and OS versions**

3.12 and latest master

**Additional context**

We have some prepared binary data that is attached to each point in the layer. The current method is to encode the data in base64 and add it as a TEXT field, then later run a separate program that reads each string, decodes it and reinserts it as a proper binary blob.

I have created a patch/PR based on the PostGis implementation of bytea since binary data has worked there since @nyalldawson's commits last year. However, to my surprise, importing didn't work there either.

![image](https://user-images.githubusercontent.com/454644/82822234-ba438100-9ea5-11ea-9bc1-cd601473f7d3.png)

However, adding the field using the add field dialog works.

![image](https://user-images.githubusercontent.com/454644/82822539-3fc73100-9ea6-11ea-84d0-5dd9f2abfb6c.png)

![image](https://user-images.githubusercontent.com/454644/82822686-78670a80-9ea6-11ea-935a-055b2f9c5afa.png)

This does not currently work in Spatialite, since it doesn't have any QVariant::ByteArray support today.

I can make a separate PR with the few lines that is necessary to make importing work in PostGIS too:

![image](https://user-images.githubusercontent.com/454644/82825327-53c16180-9eab-11ea-96d0-384db8de274f.png)

**Comment by gioman** [comment]
> I can make a separate PR with the few lines that is necessary to make importing work in PostGIS too:

@audun that would be much appreciated.

**Comment by m-kuhn** [comment]
@audun is this still valid after #36708 has been merged?
If not, this can be closed (and next time just add a `Fixes #36705` to the commit or pull request message and this will be done automatically)

**Comment by audun** [comment]
@m-kuhn Should perhaps have been two issues, but the PostGIS support was practically finished and needed a very small touch to make it work.

The issue should be fully resolved with #36744.

**Comment by audun** [comment]
Fantastic, a big thanks to the everyone involved. I'm very happy to see the PRs merged so quickly. A satisfying and pleasant process from beginning to end. 👍
